### DISCUSSION TITLE
Actor Attendants

### DISCUSSION THREAD

**Discussion by cmr9082** [AngryBeaver/beavers-crafting#77] [Ideas]
It would be nice to more than just having the attendants of the tools to also require certain Actors to within a tiles radius of the crafting actor. For example, if you were to make an actor, eg a blacksmith, he would have to be present to assist a PC with a recipe. This could also be used more for storytelling to ensure that a dm knows an npc was present when it was crafted. 

Additionally, but mainly, you could use it for workstations. For example, lets say a PC wanted to craft a little more than just a potion that you can do with a kit, but instead required an anvil, you could create an actor named anvil and mark them as present on the recipe and now a character needs to be at an anvil to craft a sword, not just have some hammers. I've loved using these sort of pseudo actors as chests, I would use it as an additional character sheet my players could access to drop stuff off in, but I would enjoy using more of these 'pseudo-actors'.

**Comment by AngryBeaver** [comment]
Well i am currently working on something that in the end would also help with that but it is a long way to go.
Actually i have read the anvil idea somewhere on foundry and since then an idea of a new module has arised.
"beavers-proximity-action."

What i am planing is a module that let you drop activities on the map and any token can interact with activities near your token. (like the anvil)

But then i worked on beavers-gamepad for my own party and this also pushed inputs into the new module. Now i primary focusing on grid and wall based activities and already have postponed the token based activity like the anvil.
(start activities will likely be " search for secret doors", "unlock doors", "find traps", "search for loot", "find clues" etc.)

There after i will improve this to token interactions. And sometime after that i would implement a proximity action extension for beavers-crafting that then would allow such things like anvils or questgiver.

But my idea for an anvil is a bit diffrent from yours.
I plan to have a dropThing (npc token)? that can have recipes of its own and this works as an proximity action so you when your actor gets close you can select an activity with the anvil to craft those recipes that are only available with the anvil.

But till then there is still a lot of work ahead and this won't come soon. And it probably won't come any other way sooner as my main activity besides of bug fixing will be invested into this new module.

**Reply by cmr9082** [reply]
I like this, honestly, to me it doesn't really matter how it works, through token based activity or radius checks. I'm just glad you have something like this planned. Its additional uses also have my mind running, thinking of roll tables for a "search the area" interaction and just in general how it could be used gets me hyped. I'm a big fan of your work, so stay motivated, cause I'm looking forward to see your progress.

**Comment by AngryBeaver** [comment]
just headsup: i am totally stucked in this module and frustrated currently there is no process for this, sry.
